crates/gpui/src/platform/windows/platform.rs

```diff
@@ -396,14 +392,110 @@ impl Platform for WindowsPlatform {
+        let _ = unsafe { EmptyClipboard() };
+
+        // allocate memory to hold clipboard contents
+        // size is wide_text.len() * 2 because each character is 2 bytes (u16)
+        let handle = unsafe { GlobalAlloc(GLOBAL_ALLOC_FLAGS(0), wide_text.len() * 2).unwrap() };
+
+        // lock the handle to get a mutable pointer to the memory
+        let locked_data = unsafe { GlobalLock(handle) };
+
+        if !locked_data.is_null() {
+            // copy clipboard contents into the memory
+            unsafe {
+                std::ptr::copy_nonoverlapping(
+                    wide_text.as_ptr(),
+                    locked_data.cast(),
+                    wide_text.len()
+                );
+            }
+
+            let result = unsafe { SetClipboardData(CF_UNICODETEXT.0.into(), HANDLE(locked_data as isize)) };
+            if result.is_err() {
+                log::error!("SetClipboardData failed: {:#?}", unsafe { GetLastError() });
+            }
+        }
+
+        unsafe {
+            GlobalUnlock(handle).log_err();
+            CloseClipboard().log_err();
+
+        }
+
     }
 
-    // todo!("windows")
     fn read_from_clipboard(&self) -> Option<ClipboardItem> {
-        unimplemented!()
+        // check if the clipboard has unicode
+        let is_clipboard_valid = unsafe { IsClipboardFormatAvailable(CF_UNICODETEXT.0.into()).is_ok() };
+        if !is_clipboard_valid {
+            return None;
+        }
+
+        // open clipboard
+        let result = unsafe { OpenClipboard(None) };
+        if result.is_err() {
+            log::error!("OpenClipboard failed: {:#?}", unsafe { GetLastError() });
+            return None;
+        }
+
+        // get clipboard data as unicode
+        let clipboard_data = unsafe { GetClipboardData(CF_UNICODETEXT.0.into()) };
+        if clipboard_data.is_err() {
+            log::error!("GetClipboardData failed: {:#?}", unsafe { GetLastError() });
+            return None;
+        }
+
+        // convert from HANDLE to HGLOBAL
+        let clipboard_data_hglobal = HGLOBAL(clipboard_data.unwrap().0 as *mut c_void);
+
+        let locked_data: *mut c_void = unsafe { GlobalLock(clipboard_data_hglobal) };
+        if locked_data.is_null() {
+            log::error!("GlobalLock failed: {:#?}", unsafe { GetLastError() });
+            return None;
+        }
+
+        // get length of string by looking for first null (\0) byte
+        let mut len = 0;
+        let mut ptr = locked_data.cast::<u16>();
+
+        while unsafe { *ptr } != 0 {
+            len += 1;
+            ptr = unsafe { ptr.offset(1) };
+        }
+
+        // convert clipboard data to a string
+        let text: Option<String> = OsString::from_wide(
+            unsafe {
+                std::slice::from_raw_parts(
+                    locked_data.cast(),
+                    // find the first null byte in the clipboard data
+                    len
+                )
+            }
+        ).into_string().ok();
```

Comment:
probably don't need `ok()` here since we are checking for `Some` afterwards. Might as well keep it as a `Result` here.